### PR TITLE
Use SplitMix

### PR DESCRIPTION
**Pull request by curiousleo**
The purpose of this PR is to use the pseudo-random number generation algorithm from [splitmix](https://hackage.haskell.org/package/splitmix) while keeping the public interface exactly as it is.

### CODE REVIEW

File: System/Random.hs [hunk 1]
```diff
@@ -564,7 +475,7 @@ getStdGen  = readIORef theStdGen
 
 theStdGen :: IORef StdGen
 theStdGen  = unsafePerformIO $ do
-   rng <- mkStdRNG 0
+   rng <- SM.initSMGen
```
Comment (lehins) [inline System/Random.hs]: I wonder how is that `theStdGen` never had `NOINLINE` pragma on it, I thought ghc warns you about this sort of thing

Comment (curiousleo) [reply System/Random.hs]: I've added `NOINLINE` now to bring this in line with https://github.com/phadej/splitmix/blob/04e74fbc8447eafeb3181934e5ec581be27fc273/src/System/Random/SplitMix.hs#L291-L293

**Comment by idontgetoutmuch** [comment]
What more needs to be done to merge this into https://github.com/idontgetoutmuch/random/pull/1 (which I think of as "The Proposal")?

**Comment by lehins** [comment]
I'll give it a good review later on tonight.

Do we need to include `shell.nix` to the repo?

**Comment by lehins** [comment]
Obviously a rebase is necessary, since there are conflicts

**Comment by idontgetoutmuch** [comment]
I think it will be easier to create a new PR

**Comment by curiousleo** [comment]
Closing in favour of https://github.com/idontgetoutmuch/random/pull/22.

**Pull request closed without merging**